src/dapp/resources/jsontemplate.json

```diff
@@ -0,0 +1,197 @@
+    }
+  },
+  // Optional
+  "settings":
+  {
+    // Optional: Stop compilation after the given stage. Currently only "parsing" is valid here
+    "stopAfter": "parsing",
+    // Optional: Sorted list of remappings
+    "remappings": [ ":g=/dir" ],
+    // Optional: Optimizer settings
+    "optimizer": {
+      // disabled by default
+      "enabled": false,
+      // Optimize for how many times you intend to run the code.
+      // Lower values will optimize more for initial deployment cost, higher
+      // values will optimize more for high-frequency usage.
+      "runs": 200
+      // Switch optimizer components on or off in detail.
+      // The "enabled" switch above provides two defaults which can be
+      // tweaked here. If "details" is given, "enabled" can be omitted.
+      // "details": {
+      //   // The peephole optimizer is always on if no details are given,
+      //   // use details to switch it off.
+      //   "peephole": true,
+      //   // The unused jumpdest remover is always on if no details are given,
+      //   // use details to switch it off.
+      //   "jumpdestRemover": true,
+      //   // Sometimes re-orders literals in commutative operations.
+      //   "orderLiterals": false,
+      //   // Removes duplicate code blocks
+      //   "deduplicate": false,
+      //   // Common subexpression elimination, this is the most complicated step but
+      //   // can also provide the largest gain.
+      //   "cse": false,
+      //   // Optimize representation of literal numbers and strings in code.
+      //   "constantOptimizer": false,
+      //   // The new Yul optimizer. Mostly operates on the code of ABI coder v2
+      //   // and inline assembly.
+      //   // It is activated together with the global optimizer setting
+      //   // and can be deactivated here.
+      //   // Before Solidity 0.6.0 it had to be activated through this switch.
+      //   "yul": false,
+      //   // Tuning options for the Yul optimizer.
+      //   "yulDetails": {
+      //     // Improve allocation of stack slots for variables, can free up stack slots early.
+      //     // Activated by default if the Yul optimizer is activated.
+      //     "stackAllocation": true,
+      //     // Select optimization steps to be applied.
+      //     // Optional, the optimizer will use the default sequence if omitted.
+      //     "optimizerSteps": "dhfoDgvulfnTUtnIf..."
+      //   }
+      // }
+    },
+    // Version of the EVM to compile for.
+    // Affects type checking and code generation. Can be homestead,
+    // tangerineWhistle, spuriousDragon, byzantium, constantinople, petersburg, istanbul or berlin
+    "evmVersion": "byzantium",
+    // Optional: Change compilation pipeline to go through the Yul intermediate representation.
+    // This is a highly EXPERIMENTAL feature, not to be used for production. This is false by default.
+    "viaIR": true,
```

Comment:
Should we set these to sane defaults and then actually use them rather than popping?

Reply:
Maybe... The template right now is mainly used to be informative and demonstrate the capabilities of the std json.

Reply:
One problem would be compatibility with older solc versions. Are options just ignored if they aren't used by solc? The `evmVersion` could be especially problematic, does solc just fill this with a default if you don't supply it?